scripts/west_commands/runners/nrf_common.py

```diff
@@ -260,15 +273,35 @@ def program_hex(self):
             if self.hex_refers_region(xip_start, xip_end):
                 qspi_erase_opt = 'ERASE_ALL'
 
+        if self.erase_uicrs and self.family in UICR_RANGES:
+            for core, range in UICR_RANGES[self.family].items():
+                self.exec_op('erasepage', defer=True, core=core, page=range[0])
+
         # What tool commands do we need to flash this target?
         if self.family == 'NRF53_FAMILY':
             # nRF53 requires special treatment due to the extra coprocessor.
             self.program_hex_nrf53(erase_arg, qspi_erase_opt)
+        elif self.family == 'NRF54H_FAMILY':
+            self.program_hex_nrf54h()
         else:
             self.op_program(self.hex_, erase_arg, qspi_erase_opt, defer=True)
 
         self.flush(force=False)
 
+    def program_hex_nrf54h(self):
+        if self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENG0_CPUAPP'):
+            core = 'NRFDL_DEVICE_CORE_APPLICATION'
+        elif self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENG0_CPURAD'):
+            core = 'NRFDL_DEVICE_CORE_NETWORK'
+        else:
+            raise NotImplementedError("Unsupported core")
+
+        if self.hex_has_uicr_content() and not self.erase_uicrs:
+            self.exec_op('erasepage', defer=True, core=core,
+                         page=UICR_RANGES[self.family][core][0])
```

Comment:
Kind of an edge case, but if app's hex somehow contains radio's UICR, then only app's UICR will be erased.

Reply:
I'd say it's a case we should avoid? To generate the radiocore UICR we'd need a cpurad build as well, so we'll have the cpurad image as well, right?

Reply:
Btw, I've done it this way in case you just rebuild cpuapp, but radiocore stays the same (and there's no resource overlap in the changes...). I added `--erase-uicrs` CLI option to force the erase of all UICRs

Reply:
Sure, it's just something we may have to be conscious of in the context of sysbuild and especially in NCS.